### PR TITLE
Add function `eventually()` for testing

### PR DESCRIPTION
**Pull request by maratori**
New function `eventually()` allows to wait for an expected state in tests. Iterator and context managers are used to be able repeating assertions.

Alternative approach is to accept function as an argument, and call it several times. But in this case user has to create a function inside test to make an assert. That's because `assert` is a statement and python doesn't allow it inside the `lambda`.

### CODE REVIEW

File: lona/pytest.py [hunk 1]
```diff
@@ -6,31 +13,38 @@
...
-        lona_app = LonaApp(project_root or request.fspath)
+        lona_app = LonaApp(project_root or str(request.fspath))
         setup_app(lona_app)
 
         # setup client
-        def setup_aiohttp_app(loop):
+        def setup_aiohttp_app(loop: AbstractEventLoop) -> Application:
             lona_app.setup_server(loop=loop)
 
+            assert lona_app.aiohttp_app is not None
```
Comment (fscherf) [inline lona/pytest.py:43]: Why is this necessary? This has nothing to todo with types also, so it should be mentioned in the commit message at least.

This assert can fail in end user test suites. Therefore it should have an assertion message

File: lona/pytest.py [hunk 2]
```diff
@@ -73,6 +87,8 @@ def setup_aiohttp_app(loop):
 
         client = await aiohttp_client(setup_aiohttp_app)
 
+        assert server is not None
```
Comment (fscherf) [inline lona/pytest.py:93]: Same here

File: lona/pytest.py [hunk 3]
```diff
@@ -96,3 +99,56 @@ def setup_aiohttp_app(loop: AbstractEventLoop) -> Application:
...
+
+        @asynccontextmanager
+        async def attempt() -> AsyncIterator[None]:
+            nonlocal context_manager_was_used, success
+            context_manager_was_used = True
+
+            try:
+                yield  # execute assertions
+
+            except KeyboardInterrupt:
```
Comment (fscherf) [inline lona/pytest.py:139]: Why are KeyboardInterrups excepted if they are reraised without further handling?

File: lona/pytest.py [hunk 4]
```diff
@@ -96,3 +99,56 @@ def setup_aiohttp_app(loop: AbstractEventLoop) -> Application:
         )
 
     return setup_lona_project_context
+
+
+def eventually(timeout: float = 5, interval: float = 1) -> Iterator[AsyncContextManager]:  # NOQA: LN001
+    """
+        Wait for expected state in async test.
```
Comment (fscherf) [inline lona/pytest.py:106]: The comment don't has to be indented this far

**Review by fscherf — CHANGES_REQUESTED** [review]
Great PR! I am really looking forward to use this. Just some minor nitpicking

File: lona/pytest.py (hunk 4 above)
Comment (maratori) [reply lona/pytest.py]: Fixed

File: lona/pytest.py (hunk 3 above)
Comment (maratori) [reply lona/pytest.py]: I thought that it is necessary to be able interrupting tests. But It turned out that I was wrong. Removed this.

File: lona/pytest.py (hunk 1 above)
Comment (maratori) [reply lona/pytest.py:43]: > Why is this necessary?

That's a good question. Because from types perspective it is `Optional[Application]`, but should be not optional.
Before `lona_app.setup_server()` it is `None` and become real application after that.

I removed assertions and used `cast` instead.

**Comment by fscherf** [comment]
@maratori Thanks for that! I will use this API in #137
